go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/containerd/containerd/v2
 
-go 1.23.0
+go 1.24.0
```

Comment:
Is go1.24 a hard requirement as minimum version by the other modules?

If it's a MUST, then we should put this on 1.24.3; go1.24 versions older than that have a known regression resulting in random SEGFAULTS

Reply:
> Is go1.24 a hard requirement as minimum version by the other modules?

Yes 1.24 is a hard requirement from k8s that we need to support a few new features in new cri-api release (listed in PR description).

> If it's a MUST, then we should put this on 1.24.3

This sounds reasonable to me, although I'm not sure if we want to use newer version than k8s though 🤔 which still uses 1.24.0: 

https://github.com/kubernetes/kubernetes/blob/a3097010faac734fb4956dbc91ae9034d0a9f840/go.mod#L9

Reply:
Updated our CI to go 1.24.3. We probably can keep the go min version in go.mod to 1.24.0? WDYT?

Reply:
I think in this case we want the minimum version to be 1.24.3 in go.mod (we can add a comment to it if needed), because we know that versions before that are incompatible, so nobody should build with versions of go1.24 before that.

Reply:
> This sounds reasonable to me, although I'm not sure if we want to use newer version than k8s though 🤔 which still uses 1.24.0

Kubernetes builds with latest patch versions, but leaves the go directive at the .0 patch release

Reply:
> I think in this case we want the minimum version to be 1.24.3 in go.mod 

Update go min version to 1.24.3 as well :)

Reply:
> Kubernetes builds with latest patch versions, but leaves the go directive at the .0 patch release

Yeah, I normally prefer `.0` as well, as in most cases "any patch release higher can be used"; this was a bit of an exception.

> Update go min version to 1.24.3 as well :)

Thanks!